src/main/java/org/squbs/sample/http/SampleHttpSvc.java

```diff
@@ -0,0 +1,100 @@
+package org.squbs.sample.http;
+
+import akka.NotUsed;
+import akka.http.javadsl.marshallers.jackson.Jackson;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import akka.util.Timeout;
+import org.squbs.sample.app.AppConstants;
+import org.squbs.sample.app.ChunkRequest;
+import org.squbs.sample.app.PingRequest;
+import org.squbs.sample.app.PingResponse;
+import org.squbs.unicomplex.AbstractRouteDefinition;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import static akka.http.javadsl.server.PathMatchers.segment;
+import static akka.pattern.PatternsCS.ask;
+
+/**
+ * The route definition.
+ */
+public class SampleHttpSvc extends AbstractRouteDefinition {
+
+    Timeout timeout = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+
+    @Override
+    public Route route() {
+
+        return route(
+           get(() -> route(anonymous, withName, chunks, route(complete(StatusCodes.BAD_REQUEST)))),
+           post(() -> path("hello", () ->
+                entity(Jackson.unmarshaller(PingRequest.class), request -> {
+
+                CompletionStage<PingResponse> respCS =
+                    ask(context().actorSelection(AppConstants.pingActorPath),
+                        request, timeout)
+                        .thenApplyAsync(PingResponse.class::cast);
+
+                return completeOK(respCS.thenApply(o -> o.message), Jackson.marshaller());
+                }))
+           ),
+            complete("Hello")
+        );
+    }
+
+    private  Route anonymous = route(
+        pathPrefix("hello", () -> pathEnd(() -> {
+            CompletionStage<PingResponse> respCS =
+                ask(context().actorSelection(AppConstants.pingActorPath),
+                    new PingRequest("anonymous"), timeout)
+                    .thenApplyAsync(PingResponse.class::cast);
+
+            return completeOKWithFutureString(respCS.thenApply(o -> o.message));
+            })
+        ));
+
+    private Route withName = route(
+        pathPrefix(segment("hello").slash(PathMatchers.segment()), who -> pathEnd(() -> {
+            CompletionStage<PingResponse> respCS =
+                ask(context().actorSelection(AppConstants.pingActorPath),
+                    new PingRequest(who), timeout)
+                    .thenApplyAsync(PingResponse.class::cast);
+
+            return completeOKWithFuture(respCS.thenApply(o -> o.message).toCompletableFuture(),
```

Comment:
Any reason to use `completeOK` vs `completeOKWithFuture ` vs `completeOKWithFutureString `.